### ISSUE TITLE
Trying to create a driver, Error 306 and hmd not found

### ISSUE THREAD

**Issue by alex9099** [ValveSoftware/openvr#566]
Hi, i'm trying to create a driver that recieves tracking info from external source (sockets server that recieves gyroscope data from android), when i keep the sample driver as it is (inside the runframe function), it works ok, but if i implement my function that is suposed to update it just fails with error 306 and the status window saying hmd not found, also i have error 103 after the "Unable to load Driver VirtualCheap from"

The code that i'm trying to use is here https://github.com/VirtualCheap/OpenVRCardBoard/tree/tracking 

Problem seems to be in this line https://github.com/VirtualCheap/OpenVRCardBoard/blob/tracking/src/driver_VirtualCheap.cpp#L334

**Comment by JoeLudwig** [comment]
Can you post a system report somewhere? The log lines around that one you included would be helpful to see.

**Comment by alex9099** [comment]
@JoeLudwig here it is https://ptpb.pw/8S7G (system report )

**Comment by JoeLudwig** [comment]
```
Fri Jun 23 2017 15:35:44.330366 - Unable to load driver VirtualCheap from /home/alexandre/.local/share/Steam/steamapps/common/SteamVR/drivers/VirtualCheap/bin/linux64/driver_VirtualCheap.so.
Fri Jun 23 2017 15:35:44.330387 - Unable to load driver VirtualCheap because of error 103. Skipping.
```

Error 103 is File Not Found. It means that our attempt to dlopen that specified file failed. Does it exist and have appropriate permissions?

This probably isn't your problem, but it's not necessary to copy your driver into the drivers directory installed by Steam. You should instead run this to register your driver with the system wherever it's normally installed:
  vrpathreg adddriver "<path>"

**Comment by alex9099** [comment]
Yes, the file exists: `-rwxr-xr-x 1 alexandre alexandre 417400 jun 23 16:01 .local/share/Steam/steamapps/common/SteamVR/drivers/VirtualCheap/bin/linux64/driver_VirtualCheap.so`

Where is that vrpathreg binary?

**Comment by JoeLudwig** [comment]
It's in bin/linux64 (or whatever platform directory.)

This is likely to be a dependency of driver_VirtualCheap.so. You can run ldd on your .so to see its dependencies.

Something that might help is to make sure that your driver is build using the Steam runtime. vrserver will run in that context, and so that's the context the driver will be loaded into.

**Comment by alex9099** [comment]
`make sure that your driver is build using the Steam runtime.`

How can i do that? (Based on the code from my repo)

Also ldd seems to find everything, the only non linked lib is the fiirst (dont remember name, not on pc right noe)

**Comment by JoeLudwig** [comment]
You can find lots of details about the Steam runtime here: https://github.com/ValveSoftware/steam-runtime

**Comment by alex9099** [comment]
yea, now i'm thinking, i'm using arch, which has up to date libraries, that may be a problem :/ i'll check that steam-runtime

**Comment by alex9099** [comment]
oh... arch does not have apt-get this seems to be a problem... grrr, isnt there a premade chroot that does not need to use apt-get?

**Comment by alex9099** [comment]
@JoeLudwig The code i have is OK? can you try to compile it on ubuntu?

**Comment by alex9099** [comment]
when i do ./vrcmd --pollpose (assuming that this exists, since there is ./vrcmd --poolcontrollers) i get VRInitError_IPC_ConnectFailed

**Comment by alex9099** [comment]
@JoeLudwig 

Here is output of ldd driver_VirtualCheap.so 
```        linux-vdso.so.1 (0x00007fffce074000)
        libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007feb43ceb000)
        libm.so.6 => /usr/lib/libm.so.6 (0x00007feb439d9000)
        libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007feb437c2000)
        libc.so.6 => /usr/lib/libc.so.6 (0x00007feb4341d000)
        /usr/lib64/ld-linux-x86-64.so.2 (0x0000561310ec0000)
```

**Comment by alex9099** [comment]
Another thing (what a big monologue): i think the problem is not with libs, because if i compile the sample driver openvr/steamvr works fine, unless it has something to do with multiple file compiler stuff

**Comment by alex9099** [comment]
@godbyk can you give me some help about this?
